com.playeveryware.eos/Runtime/Core/Config.cs

```diff
@@ -324,19 +303,51 @@ protected virtual async Task ReadAsync()
         /// </summary>
         protected virtual void Read()
         {
-            // This conditional exists because writing a config file is only
-            // something that should ever happen in the editor.
-            // This is the config writing for Editor Playmode
-            // Use WriteAsync instead on Editor Not-Playmode
+            EnsureConfigFileExists(false);
+            _lastReadJsonString = FileSystemUtility.ReadAllText(FilePath);
+            JsonUtility.FromJsonOverwrite(_lastReadJsonString, this);
+        }
+
+        /// <summary>
+        /// Determines if the config file exists, and if it does not, and the
+        /// editor is running, then create the file.
+        /// </summary>
+        /// <param name="isAsync">
+        /// Indicates whether the operation should run synchronously or
+        /// asynchronously.
+        /// </param>
+        /// <returns>Task.</returns>
+        private async Task EnsureConfigFileExistsAsync(bool isAsync)
+        {
+            bool fileExists = isAsync ? await FileSystemUtility.FileExistsAsync(FilePath) : FileSystemUtility.FileExists(FilePath);
+
 #if UNITY_EDITOR
-            if (!File.Exists(FilePath)) 
+            if (!fileExists)
             {
-                Write();
+                if (isAsync)
+                    await WriteAsync();
+                else
+                    Write();
             }
+#else
+            // If the editor is not running, then the config file not
+            // existing should throw an error.
+            throw new FileNotFoundException(
+                $"Config file \"{FilePath}\" does not exist.");
 #endif
+        }
 
-            _lastReadJsonString = FileUtility.ReadAllText(FilePath);
-            JsonUtility.FromJsonOverwrite(_lastReadJsonString, this);
+        /// <summary>
+        /// Determines if the config file exists, and if it does not, and the
+        /// editor is running, then create the file.
+        /// </summary>
+        /// <param name="isAsync">
+        /// Indicates whether the operation should run synchronously or
+        /// asynchronously.
+        /// </param>
+        private void EnsureConfigFileExists(bool isAsync)
```

Comment:
It feels odd that this allows for `isAsync`, given that it always will be synchronous since it's getting the awaiter and result.

Reply:
Per [this comment](https://github.com/PlayEveryWare/eos_plugin_for_unity/pull/864#discussion_r1735448207), I am marking _this_ conversation as resolved.